File: src/iodr/schema.py
from datetime import datetime
from typing import Annotated
from pydantic import BaseModel, Field


class TransactionRequest(BaseModel):
    user_id: str
    amount: Annotated[float, Field(gt=0, description="Amount should be greater than 0")]


class Transaction(TransactionRequest):
    transaction_id: str
    created_at: datetime


class User(BaseModel):
    user_id: str
    transactions: list[Transaction]
    total_amount: float
    number_of_transactions: int

File: ui/src/schema.ts
export interface Transaction {
    transaction_id: string;
    user_id: string;
    amount: number;
    created_at: string;
}

export interface User {
    user_id: string;
    transactions: Transaction[];
    total_amount: number;
    number_of_transactions: number;
}

export interface UserSummary extends User { }

export type UserSummaryResponse = [Transaction[], number, number];
